### ISSUE TITLE
Should we used norm or squared norm in loss function

### ISSUE THREAD

**Issue by manuelli** [RobotLocomotion/pytorch-dense-correspondence#46]
Currently we are using squared error. Changing this could affect learning rates, and other params.

Tanner uses squared loss. Slightly different from ours though.

- matches D(I_a, I_b, u_a, u_2)^2
- non-matches max(0, M - D(I_a, I_b, u_a, u_b))^2
